crates/pessimistic-proof/src/lib.rs

```diff
@@ -2,8 +2,8 @@ pub mod keccak;
 pub mod local_exit_tree;
 
 mod proof;
+pub use alloy_primitives::{Address, Signature, U256};
 pub use proof::{generate_pessimistic_proof, PessimisticProofOutput, ProofError};
-pub use reth_primitives::{Address, Signature, U256};
```

Comment:
Note: `alloy_primitives::Signature` has a different serde-based serialization format from `reth_primitives::Signature`.

This affects:
* the hex-serialization of `Certificate` used for cert submission
* the format of Certificate as stored in the database
* the serialization of signature passed in `MultiBatchHeader` to the zkvm executor via stdin

Reply:
Added a compatibility layer so the serialization format remains unchanged. It works by introducing a crate `agglayer_primitives` which is used in place of `reth_primitives`. It re-exports a bunch of types from `alloy_primitives` but for signature it introduces a new one. Internally `agglayer_primitives::Signature` uses `alloy_primitives::PrimitiveSignature` but for (de)serialization, it uses the same format as `reth_primitives::Signature`.